tracker/src/tracker-control/tracker-control-general.c

```diff
@@ -535,15 +535,18 @@ get_uid_for_pid (const gchar  *pid_as_string,
 	GFileInfo *info;
 	GError *error = NULL;
 	gchar *fn;
+	gchar *proc_dir_name;
 	guint uid;
 
+    proc_dir_name = g_build_filename ("/proc", pid_as_string, NULL);
+
 #ifdef __sun /* Solaris */
-	fn = g_build_filename ("/proc", pid_as_string, "psinfo", NULL);
+	fn = g_build_filename (proc_dir_name, "psinfo", NULL);
 #else
-	fn = g_build_filename ("/proc", pid_as_string, "cmdline", NULL);
+	fn = g_build_filename (proc_dir_name, "cmdline", NULL);
 #endif
 
-	f = g_file_new_for_path (fn);
+	f = g_file_new_for_path (proc_dir_name);
 	info = g_file_query_info (f,
 	                          G_FILE_ATTRIBUTE_UNIX_UID,
```

Comment:
Since we are querying for "proc_dir_name", the error below should use also "proc_dir_name" and not "fn".